R/control_chart.R

```diff
@@ -0,0 +1,139 @@
+#' Create a control chart
+#' 
+#' Create a control chart, aka Shewhart chart:
+#' \url{https://en.wikipedia.org/wiki/Control_chart}. 
+#'
+#' @param d data frame or a path to a csv file that will be read in
+#' @param measure variable of interest mapped to y-axis
+#' @param x variable to go on the x-axis, often a time variable. If unspecified
+#' row indices will be used
+#' @param group1 Optional grouping variable to be panelled horizontally
+#' @param group2 Optional grouping variable to be panelled vertically
+#' @param center_line Function used to calculate central tendency. 
+#' Defaults to mean
+#' @param sigmas Number of standard deviations above and below the central 
+#' tendency to call a point influenced by "special cause variation."
+#' Defaults to 3
+#' @param save_to Optional file path to save chart. If not provided, the chart
+#' will be printed on screen
+#' @param plot_width In inches. Only used if save_to is specified
+#' @param plot_height In inches. Only used if save_to is specified
+#' @param plot_title Title in upper-left
+#' @param plot_catpion Caption in lower-right
+#' @param plot_font_size Base font size; text elements will be scaled to this
+#'
+#' @return Generally called for the side effect of printing the control chart
+#' or writing the control chart to file, depending on whether save_to is 
+#' specified. Invisibly, returns a ggplot object for further customization.
+#' @export
+#' @import ggplot2
+#' @importFrom readr read_csv
+#' @importFrom tibble data_frame
+#' @importFrom stats as.formula
+#'
+#' @examples
+#' d <- 
```

Comment:
Some comments and spacing in the most basic example would be helpful. `# Generate data`... On that note, maybe we could standardize to one simple and one advanced example per file?

Reply:
@mmastand I'll add some comments. I think the number of examples a function calls for depends quite a bit on how customizable the function is.